metrics/config.go

```diff
@@ -341,6 +356,27 @@ func prometheusPort() (int, error) {
 	return int(pp), nil
 }
 
+// prometheusHost returns the host configured via the environment
+// for the Prometheus metrics exporter if it's set, a default value otherwise.
+// Validation is performed via parsing the host value as part of a url.
+func prometheusHost() (*string, error) {
+	phStr := os.Getenv(prometheusHostEnvName)
+	if phStr == "" {
+		return ptr.String(defaultPrometheusHost), nil
+	}
+	u, err := url.Parse(phStr)
+	if err != nil {
+		return nil, fmt.Errorf("the environment variable %q could not be parsed as a host: %w",
+			prometheusHostEnvName, err)
+	}
+	host, _, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract host from %q: %w",
+			u.Host, err)
+	}
+	return &host, nil
```

Comment:
https://golang.org/pkg/net/url/#URL.Hostname perhaps?

Reply:
This will not work if you pass an ip btw, it returns "".